chimg/config.py

```diff
@@ -3,20 +3,41 @@
 
 import pathlib
 from typing import List, Optional
-from pydantic import BaseModel, Field
+from pydantic import BaseModel, Field, model_validator
 
 
 class ConfigFile(BaseModel):
     """
-    A file configuration
+    A file configuration.
+
+    This allows for easily creating files on the target system either with content or by copying a file or directory.
+
+    If `content` is provided, the file will be created with the content provided.
+
+    If `source` is provided and the source is a file, the file will be copied to the destination path.
+
+    If `source` is provided and the source is a directory, all files in the directory will be copied to the
+    destination path (destination will be treated as a directory).
     """
 
-    destination: str
-    content: str
+    destination: str = Field(description="Destination file or directory path")
+    content: Optional[str] = Field(description="Content of the file being created", default=None)
+    source: Optional[str] = Field(description="Source file or directory for local uploads", default=None)
```

Comment:
I would not use the term "local uploads". imo that's missleading. maybe just "Source file or source directory to copy into destination` ?